packages/lib-utils/src/app/redux/ReduxExtensionProvider.tsx

```diff
@@ -0,0 +1,27 @@
+import { useResolvedExtensions, isReduxProvider } from '@openshift/dynamic-plugin-sdk';
+import type { ReduxProvider } from '@openshift/dynamic-plugin-sdk';
+import * as React from 'react';
+import { Provider } from 'react-redux';
+
+/**
+ * Consumes the core.redux-
+ */
+const ReduxExtensionProvider: React.FC = ({ children }) => {
+  const [reduxProviderExtensions, reduxProvidersResolved] =
+    useResolvedExtensions<ReduxProvider>(isReduxProvider);
+
+  return reduxProvidersResolved ? (
+    <>
+      {reduxProviderExtensions.reduce(
+        (c, e) => (
+          <Provider store={e.properties.store} context={e.properties.context}>
+            {c}
+          </Provider>
+        ),
+        children,
+      )}
+    </>
+  ) : null;
```

Comment:
There are currently two scenarios which may cause extensions to be put into or out of use at any point in time:
- change in a plugin's enabled status
- change in the feature flags, which in turn impacts the availability of individual extensions

Extensions in use are computed as all extensions from plugins that are enabled, which also meet the feature flag criteria according to the current feature flag values.

Therefore,
```ts
useResolvedExtensions<ReduxProvider>(isReduxProvider)
```
may cause this hook to be re-run multiple times, providing different results on each run.

I think we need to guard against this, otherwise we may "un-render" and then "re-render" a Redux `Provider` which would effectively reset it (?)

Reply:
The context and store should be the same instance on a re-render. So even if the provider was re-rendered, those who rely on the redux store would still be accessing the same redux context and store.

Reply:
One more question, if a `ReduxProvider` extension is put out of use (due to change in flags or due to the owner plugin getting disabled) but some component down in the tree still attempts to access the associated context or store, the responsibility to handle this is on the consuming component, correct?

Reply:
Correct. If a nested component relies on redux and the provider is not present, it's their responsibility. Under normal circumstances the system will ensure its presence in time for consumption.

A redux provider that is disabled by a flag, while possible, this scenario is less likely to happen as the presence would have a no-op effect if unused and therefore using a flag wouldn't have any real benefit.

If the plugin is disabled, then all contributions from said plugin would also be disabled and therefore there would be no reliance on the provider.